### PR TITLE
Deselect other state fields when editing.

### PR DESCRIPTION
**Pull request by ghost**
OK to merge.

Fixes https://app.asana.com/0/743996399535511/757707017946664

### CODE REVIEW

File: packages/haiku-creator/src/react/components/StateInspector/StateRow.js [hunk 1]
```diff
@@ -281,18 +279,24 @@ class StateRow extends React.Component {
       <form key={`${this.props.stateName}-state`}
         onMouseOver={() => this.setState({isHovered: true})}
         onMouseOut={() => this.setState({isHovered: false})}>
-        {!this.state.isEditing && !this.props.isNew
+        {!this.props.isEditing && !this.props.isNew
           ? <div style={STYLES.stateWrapper}>
             <div
-              onDoubleClick={() => this.setState({isEditing: true, didEscape: false, editingTarget: 'name'})}
```
Comment (roperzh) [inline packages/haiku-creator/src/react/components/StateInspector/StateRow.js:287]: Could you please move this and the callback below into a constructor function? for the why and details: https://github.com/HaikuTeam/mono/pull/627#discussion_r204841886

File: packages/haiku-creator/src/react/components/StateInspector/StateInspector.js [hunk 2]
```diff
@@ -188,12 +189,12 @@ class StateInspector extends React.Component {
 
   openNewStateForm () {
     if (!this.state.addingNew) {
-      this.setState({addingNew: true});
+      this.setState({addingNew: true, editingStateName: 'new-row'});
```
Comment (roperzh) [inline packages/haiku-creator/src/react/components/StateInspector/StateInspector.js]: nit: could you please make an enum with all state names and compare based on that? something like:

```js
const editingStates = {
  NOT_EDITING: 0,
  NEW_ROW: 1,
}
```

**Review by roperzh — COMMENTED** [review]
LGTM! only nit functionality wise: if you click the `+` button twice there's a small flicker, but not a big deal

File: packages/haiku-creator/src/react/components/StateInspector/StateInspector.js (hunk 2 above)
Comment (ghost) [reply packages/haiku-creator/src/react/components/StateInspector/StateInspector.js]: These aren't actually the states, it's the name of the state property that the user defined, with the exception of `new-row` which was set by someone on line 240.

Maybe `editingStateName` isn't a good variable name, but my intention was for it to be `null` when the user is not editing anything, and `"myCustomStateProp"` when the user is editing `myCustomStateProp`.

Comment (roperzh) [reply packages/haiku-creator/src/react/components/StateInspector/StateInspector.js]: ah, that makes sense, feel free to disregard this

**Comment by ghost** [comment]
@roperzh I simplified the state and moved those callbacks into the constructor, let me know if that looks right to you.

I think the flicker is caused by `handleClickOutside`, which is a can of worms I don't feel like opening right now...

**Comment by roperzh** [comment]
LGTM! :shipit:

**Pull request merged**